src/Pact/Types/Crypto.hs

```diff
@@ -219,64 +168,60 @@ instance ConvertBS Ed25519.Signature where
   fromBS = Right . Ed25519.Sig
 #endif
 
-
-
---------- SCHEME ETH INSTANCES --------
-
-instance Scheme (SPPKScheme 'ETH) where
-  type PublicKey (SPPKScheme 'ETH) = ECDSA.PublicKey
-  type PrivateKey (SPPKScheme 'ETH) = ECDSA.PrivateKey
-  type Signature (SPPKScheme 'ETH) = ECDSA.Signature
-
-
-  _sign _ (Hash msg) pub priv = ECDSA.signETH (fromShort msg) pub priv
-  _valid _ (Hash msg) pub sig = ECDSA.validETH (fromShort msg) pub sig
-  _genKeyPair _ = ECDSA.genKeyPair
-  _getPublic _ = Just . ECDSA.getPublicKey
-  _formatPublicKey _ p = ECDSA.formatPublicKeyETH (toBS p)
-
-
-
-instance ConvertBS (ECDSA.PublicKey) where
-  toBS = ECDSA.exportPublic
-  fromBS s = maybeToEither ("Invalid ECDSA Public Key: " ++ show (toB16Text s))
-             (ECDSA.importPublic s)
-instance ConvertBS (ECDSA.PrivateKey) where
-  toBS = ECDSA.exportPrivate
-  fromBS s = maybeToEither ("Invalid ECDSA Private Key: " ++ show (toB16Text s))
-             (ECDSA.importPrivate s)
-instance ConvertBS (ECDSA.Signature) where
-  toBS = ECDSA.exportSignature
-  fromBS s = maybeToEither ("Invalid ECDSA Signature: " ++ show (toB16Text s))
-             (ECDSA.importSignature s)
-
-
-
+instance Scheme (SPPKScheme 'WebAuthn) where
+
+  _valid _ msg (PubBS pubBS) (SigBS sigBS) =
+    case runVerification of
+      Left _ -> False
+      Right () -> True
+    where
+      -- Verifying a WebAuthn signature requires that we know the payload
+      -- signed by the WebAuthn keys on the client device. This payload is
+      -- a combination of the Challenge (in our case, the PactHash of a
+      -- transaction), a JSON object of "client metadata", and data about
+      -- the authenticator hardware.
+      -- We require this data to be part of our WebAuthn signature so that
+      -- we can reconstitute the payload that was signed on the browser.
+      runVerification :: Either String ()
+      runVerification = do
+
+        -- Decode our WebAuthn signature object from a `UserSig` string.
+        WebAuthnSignature
+          { authenticatorData
+          , signature
+          , clientDataJSON } <- A.eitherDecode (BSL.fromStrict sigBS)
```

Comment:
Maybe 

```suggestion
        WebAuthnSignature{..} <- A.eitherDecode (BSL.fromStrict sigBS)
```

using  `{-# LANGUAGE RecordWildCards #-}`

Reply:
I like NamedFieldPuns because it makes the identifiers being brought into scope explicit. Meaning: if someone reading the code didn't know what `authenticationData` is, or where it comes from, `NamedFieldPuns` makes it greppable, while `RecordWildCards` doesn't.

Reply:
Alternatively, construct the thing directly since we're not even really punning as much as we are constructing directly.

Reply:
I like NamedFieldPuns even for vanilla record desructuring too, since it removes the possibility of accidentally naming fields out of order. Meaning, I could write:

```
  WebAuthnSignature sig authData clientData <- decode sigBS
    -- (Is this what you mean by destruct the thing directly?)
  ...
``` 

and it would typecheck but be subtly wrong, because I accidentally swapped the field names around for fields that have the same type.